File: python/src/acedatacloud/resources/platform.py
"""Management-plane resources (applications, credentials, models, config)."""

from __future__ import annotations

from typing import Any


class _Applications:
    def __init__(self, transport: Any) -> None:
        self._transport = transport

    def list(self, **params: Any) -> dict[str, Any]:
        return self._transport.request("GET", "/api/v1/applications/", params=params, platform=True)

    def create(self, *, service_id: str, **kwargs: Any) -> dict[str, Any]:
        body = {"service_id": service_id, **kwargs}
        return self._transport.request("POST", "/api/v1/applications/", json=body, platform=True)

    def get(self, application_id: str) -> dict[str, Any]:
        return self._transport.request("GET", f"/api/v1/applications/{application_id}/", platform=True)


class _AsyncApplications:
    def __init__(self, transport: Any) -> None:
        self._transport = transport

    async def list(self, **params: Any) -> dict[str, Any]:
        return await self._transport.request("GET", "/api/v1/applications/", params=params, platform=True)

    async def create(self, *, service_id: str, **kwargs: Any) -> dict[str, Any]:
        body = {"service_id": service_id, **kwargs}
        return await self._transport.request("POST", "/api/v1/applications/", json=body, platform=True)

    async def get(self, application_id: str) -> dict[str, Any]:
        return await self._transport.request("GET", f"/api/v1/applications/{application_id}/", platform=True)


class _Credentials:
    def __init__(self, transport: Any) -> None:
        self._transport = transport

    def list(self, **params: Any) -> dict[str, Any]:
        return self._transport.request("GET", "/api/v1/credentials/", params=params, platform=True)

    def create(self, *, application_id: str, **kwargs: Any) -> dict[str, Any]:
        body = {"application_id": application_id, **kwargs}
        return self._transport.request("POST", "/api/v1/credentials/", json=body, platform=True)

    def rotate(self, credential_id: str) -> dict[str, Any]:
        return self._transport.request("POST", f"/api/v1/credentials/{credential_id}/rotate/", platform=True)

    def delete(self, credential_id: str) -> dict[str, Any]:
        return self._transport.request("DELETE", f"/api/v1/credentials/{credential_id}/", platform=True)


class _AsyncCredentials:
    def __init__(self, transport: Any) -> None:
        self._transport = transport

    async def list(self, **params: Any) -> dict[str, Any]:
        return await self._transport.request("GET", "/api/v1/credentials/", params=params, platform=True)

    async def create(self, *, application_id: str, **kwargs: Any) -> dict[str, Any]:
        body = {"application_id": application_id, **kwargs}
        return await self._transport.request("POST", "/api/v1/credentials/", json=body, platform=True)

    async def rotate(self, credential_id: str) -> dict[str, Any]:
        return await self._transport.request("POST", f"/api/v1/credentials/{credential_id}/rotate/", platform=True)

    async def delete(self, credential_id: str) -> dict[str, Any]:
        return await self._transport.request("DELETE", f"/api/v1/credentials/{credential_id}/", platform=True)


class _Models:
    def __init__(self, transport: Any) -> None:
        self._transport = transport

    def list(self, **params: Any) -> dict[str, Any]:
        return self._transport.request("GET", "/api/v1/models/", params=params, platform=True)


class _AsyncModels:
    def __init__(self, transport: Any) -> None:
        self._transport = transport

    async def list(self, **params: Any) -> dict[str, Any]:
        return await self._transport.request("GET", "/api/v1/models/", params=params, platform=True)


class _Config:
    def __init__(self, transport: Any) -> None:
        self._transport = transport

    def get(self) -> dict[str, Any]:
        return self._transport.request("GET", "/api/v1/config/", platform=True)


class _AsyncConfig:
    def __init__(self, transport: Any) -> None:
        self._transport = transport

    async def get(self) -> dict[str, Any]:
        return await self._transport.request("GET", "/api/v1/config/", platform=True)


class Platform:
    """Synchronous management-plane client."""

    def __init__(self, transport: Any) -> None:
        self.applications = _Applications(transport)
        self.credentials = _Credentials(transport)
        self.models = _Models(transport)
        self.config = _Config(transport)


class AsyncPlatform:
    """Async management-plane client."""

    def __init__(self, transport: Any) -> None:
        self.applications = _AsyncApplications(transport)
        self.credentials = _AsyncCredentials(transport)
        self.models = _AsyncModels(transport)
        self.config = _AsyncConfig(transport)

File: typescript/src/resources/platform.ts
/** Management-plane resources. */

import { Transport } from '../runtime/transport';

class Applications {
  constructor(private transport: Transport) {}

  async list(params?: Record<string, string>): Promise<Record<string, unknown>> {
    return this.transport.request('GET', '/api/v1/applications/', { params, platform: true });
  }

  async create(opts: { serviceId: string; [key: string]: unknown }): Promise<Record<string, unknown>> {
    const { serviceId, ...rest } = opts;
    return this.transport.request('POST', '/api/v1/applications/', {
      json: { service_id: serviceId, ...rest },
      platform: true,
    });
  }

  async get(applicationId: string): Promise<Record<string, unknown>> {
    return this.transport.request('GET', `/api/v1/applications/${applicationId}/`, { platform: true });
  }
}

class Credentials {
  constructor(private transport: Transport) {}

  async list(params?: Record<string, string>): Promise<Record<string, unknown>> {
    return this.transport.request('GET', '/api/v1/credentials/', { params, platform: true });
  }

  async create(opts: { applicationId: string; [key: string]: unknown }): Promise<Record<string, unknown>> {
    const { applicationId, ...rest } = opts;
    return this.transport.request('POST', '/api/v1/credentials/', {
      json: { application_id: applicationId, ...rest },
      platform: true,
    });
  }

  async rotate(credentialId: string): Promise<Record<string, unknown>> {
    return this.transport.request('POST', `/api/v1/credentials/${credentialId}/rotate/`, { platform: true });
  }

  async delete(credentialId: string): Promise<Record<string, unknown>> {
    return this.transport.request('DELETE', `/api/v1/credentials/${credentialId}/`, { platform: true });
  }
}

class Models {
  constructor(private transport: Transport) {}

  async list(params?: Record<string, string>): Promise<Record<string, unknown>> {
    return this.transport.request('GET', '/api/v1/models/', { params, platform: true });
  }
}

class Config {
  constructor(private transport: Transport) {}

  async get(): Promise<Record<string, unknown>> {
    return this.transport.request('GET', '/api/v1/config/', { platform: true });
  }
}

export class Platform {
  readonly applications: Applications;
  readonly credentials: Credentials;
  readonly models: Models;
  readonly config: Config;

  constructor(transport: Transport) {
    this.applications = new Applications(transport);
    this.credentials = new Credentials(transport);
    this.models = new Models(transport);
    this.config = new Config(transport);
  }
}
